91 project card rework

vsheo opened the pull request:
## What does this change?

Resolves issue #91 

In this branch I redesigned the project cards and made them responsive with container queeries.


## How Has This Been Tested?
The previous version was tested thoroughly. I made only minimal changes to the HTML and verified that everything still works the same.
The changes I did make are related to the layout and the media/container queries.

## Images/videos
https://github.com/user-attachments/assets/c172b9c7-7810-4afd-9271-6070b4dccd01

## How to review
Please confirm the project cards have the exact same functionality as the previous version (on dev branch), keyboard & screenreader accessible and remains fully responsive

vsheo committed:
redesign + @container start #91

vsheo committed:
container queeries en grid layout #91

vsheo committed:
fully responsive with container queries #91

vsheo committed:
remove necessary comments/code #91

vsheo committed:
elipsis on small card #91

vsheo committed:
clean up before PR #91

Mikiyas-hs commented:
Hi Viresh i have reviewed your new version of the cards: 

great accessibilty score:
<img width="525" height="700" alt="image" src="https://github.com/user-attachments/assets/9d596fd6-2b2f-4ca4-8f0b-1cbd55edd573" />

you can easily tab threw the cards:

https://github.com/user-attachments/assets/9446da10-adc1-44bc-af7f-40067f3a8f48

the only thing you should fix is this part of your component:

<img width="1801" height="726" alt="image" src="https://github.com/user-attachments/assets/55880172-8970-4188-8ac9-5f6c82803ae0" />

as you can see are the cards not completely responsive it seems like when your title is 2 rows your card will get longer let me know if you need help fixing it maybe this article helps:
https://medium.com/@mahamudh472/how-to-fix-long-name-on-product-card-text-issue-f1f848776368

vsheo committed:
feedback PR #115

vsheo committed:
height fix #91

vsheo committed:
remove comments #91

vsheo commented:
Thank you for the feedback. I’ve processed it. I couldn’t see the text being cut off on my screen, but I think I’ve fixed it now. Could you check it again for me?

Regarding the height issue: this happens because the title is too long. I don’t want to set a fixed height, because if the product owner adds an even longer title, the card wouldn’t have enough space.
I added a min-height, but that also increases the whitespace on the card next to it when the title is long:

<img width="2120" height="674" alt="image" src="https://github.com/user-attachments/assets/024cf460-e19a-4639-b602-ff2831fbd5ce" />

This is the longest title we currently have:
<img width="2125" height="501" alt="image" src="https://github.com/user-attachments/assets/8cebf90b-e546-4bd3-b3e4-01e9d3d5917c" />

For now it looks alright, but I think we should consider switching to a single-column layout. That way it won’t matter how long the current or future titles are, the layout will always look consistent.
example:

https://github.com/user-attachments/assets/76d1fa85-e581-4c00-924d-fe14b9d712c0

Mikiyas-hs commented:
Its still the same lets merge it for now because this is the only problem you have but make a issue out of it so we could get back to this issue and fix it but for now everything seems to look good except for that small issue so i would say that you can merge it

vsheo commented:
Ive made issue #118 to get back to this problem
